tests/bgp/test_bgp_speaker.py

```diff
@@ -62,11 +62,76 @@ def skip_dualtor(tbinfo):
     """Skip running `test_bgp_speaker` over dualtor."""
     pytest_require("dualtor" not in tbinfo["topo"]["name"], "Skip 'test_bgp_speaker over dualtor.'")
 
+def dut_bgp_asn_update_status(duthost, bgp_speaker_asn, dut_asn, Asntype):
+    bgpvacstatus = "default"
+    if Asntype == "2byte" or Asntype == "4byte":
+        logger.info("Updating BGPVac ASN to 4 byte")
+        logger.info("Bgp_speaker_asn =%s" % bgp_speaker_asn)
+        logger.info("T0 ASN = %s" % dut_asn)
+        bgpvacstatus = "fourbyte"
+    duthost.command("vtysh -c \"conf t\" \
+            -c \"router bgp %s\" i\
+            -c \"neighbor BGPSLBPassive remote-as %s\" \
+            -c \"neighbor BGPSLBPassive activate\" \
+            -c \"neighbor BGPVac remote-as %s\" \
+            -c \"neighbor BGPVac activate\" \
+            -c \"exit\"" % (dut_asn, bgp_speaker_asn, bgp_speaker_asn))
+    
+    return bgpvacstatus
+
+def dut_config_change(duthost, dut_4basn):
+    duthost.shell("sudo cp /etc/sonic/config_db.json /etc/sonic/config_db_org.json")
+    bgp_config = duthost.shell("sonic-cfggen -d  --var-json 'DEVICE_METADATA'")['stdout']
+    bgp_config_json = json.loads(bgp_config)
+    for _, config in bgp_config_json.items():
+        config['bgp_asn'] = dut_4basn
+    bgp_config_json = {"DEVICE_METADATA": bgp_config_json}
+    logger.info(bgp_config_json)
+    TMP_FILE = "/tmp/bgp_config.json"
+    with open(TMP_FILE, "w") as f:
+        json.dump(bgp_config_json, f)
+
+    duthost.copy(src=TMP_FILE, dest=TMP_FILE)
+    duthost.shell("sonic-cfggen -j {} -w".format(TMP_FILE))
+    time.sleep(10)
+    duthost.shell("config save -y")
+    time.sleep(10)
+    is_config_applied=duthost.shell("sudo config reload -y")
+    time.sleep(20)
+    logger.info(is_config_applied)
+    logger.info("wait for configuration to be applied")
+    time.sleep(30)
+    updated_asn=duthost.shell("show ip bgp sum")
+    logger.info("New T0 ASN = %s" % updated_asn)
+    result="false"
+    for item in updated_asn['stdout_lines']:
+        if dut_4basn in item:
+            result= "true"
+            break
 
-@pytest.fixture(scope="module")
-def common_setup_teardown(duthosts, rand_one_dut_hostname, ptfhost, localhost, tbinfo):
+    return result
+
+def dut_config_reset(duthost, dut_asn_default):
+    duthost.shell("sudo cp /etc/sonic/config_db_org.json /etc/sonic/config_db.json")
+    duthost.shell("sudo config reload -y")
+    time.sleep(10)
+    logger.info("wait for configuration to be applied")
+    time.sleep(40)
```

Comment:
The above 3 lines can be safely replaced with a call to wait_critical_processes

Example: https://github.com/Azure/sonic-mgmt/blob/master/tests/platform_tests/test_reboot.py#L74

Reply:
taken care. Thank you